### DISCUSSION TITLE
Sanctum Headings

### DISCUSSION THREAD

**Discussion by jdanielmourao** [jdanielmourao/obsidian-sanctum#87] [General]
Hello! 🙂
I'm thinking about reworking headings on Sanctum, but first I'd like to know how you've been enjoying them, what don't you like about them, and what you want to see added.
Do you use the top headings? Do you think they're way too big?
Do you like the fact that they adapt size based on window size?
Do you wish there was more variance in sizing, instead of all the weight changes?

These are just some of the things I'm curious to hear your feedback on.

**Comment by imed-ghomari** [comment]
I always enable productive mode headings, the default is too big for me. As for the size adjustments, I think a slider would be better than the dropdown options (to be more precise).

**Reply by jdanielmourao** [reply]
I initially thought about making it a slider, but style settings doesn't currently let me customize the stops to match the typographic scale I've set for the theme. 😔

**Comment by eleanorkonik** [comment]
I also always enable productive mode headings, and also feel like the default heading fonts are huge. I personally have trouble telling "at a glance" what "level" different headers are just by size/font weight, even with the typora-style anchors (which throw me off because I have to actually _read_ them and also because they cause my headings to not be aligned with my text and thus throw off my sense of where "left align" is for the headers). 

I don't care if what you go with is colored underlines or background colors, but I personally find it overwhelming when I have to personally select a color for a bunch of settings using droplets and stuff, I don't have a good sense of what colors "look good" especially with a particular theme, I just want to easily be able to visually distinguish between different heading levels (especially 1-4) without the "smaller" headings suddenly being a totally different styling like bold or a wildly "off" color pattern. 

Before live preview became a thing I used, I just relied on colors in preview mode and the `###` stuff in edit mode, but now I'm super disoriented all the time because the codes I've found for adding background colors break in sanctum in live edit mode because they overlap too badly and make the heading text unreadable. I might need to try just doing a thick underline in different colors but haven't had a chance.

**Reply by dummifiedme** [reply]
You actually said whatever I wanted to say, but a  lot better :smile:.
+1 for every word she said.

- Default fonts seems too big. 
- Always on productive mode.
- I use the background colour
- The heading tags throw me off. 


I would just like to add one thing. The gap(top margin) between two headers in succession is zero currently. It would be great if we could have an option or some default value. Currently, I have a snippet to do it for me.

**Comment by kmaustral** [comment]
Thanks for the chance to talk about this. 

I also find the headings too big. I rarely use 1 or 2 because of the space they take up. I would prefer a smaller size with a maker, like a line under the heading for 1. 

The heading code in the preview mode is also something I would turn off if I had the option. I would prefer a cleaner look, especially as we will be spending more time in preview.

**Reply by jdanielmourao** [reply]
What do you mean by preview mode code? Do you mean the Typora-like indicators (h1,h2,etc) in Live Preview? There's a style settings option to disable that.

**Comment by sharfaroz7** [comment]
I was half minded about sanctum till I discovered the productive mode headings. I never went back to the huge headings.

**Comment by jdanielmourao** [comment]
Alright, so I've also found myself not using the top 2 headings all that much recently, and wanted to gage if people used them, which doesn't seem to be the case.
Having said that, I'm thinking about making the current productive headings into the default, and deprecating the variable/expressive ones. I've also changed heading indicators to only show up in the active line, when editing the heading itself. Here's a preview:
![AI7ZqCPkf2](https://user-images.githubusercontent.com/91087143/147879267-8c6d546d-af26-4c9a-9d5a-c5aa6ce94bbc.gif)

**Reply by dummifiedme** [reply]
> Alright, so I've also found myself not using the top 2 headings all that much recently, and wanted to gage if people used them, which doesn't seem to be the case,

I do use the top two headings as I am using the file name sync plugin. Just saying :D

**Comment by jdanielmourao** [comment]
Just released Sanctum v0.5.5, with reworked Headings!
Let me know what you think. 🙂

**Reply by imed-ghomari** [reply]
I'm new to this theme, so I don't know if the color options have just been added or were there before, but I really like what you picked. And I think that you'll get more users with the new default heading size. I myself didn't want to use your theme at first because of the headings, so new inexperienced users will certainly enjoy the change.

**Reply by jdanielmourao** [reply]
Hi! Hope you enjoy Sanctum! :) 
The colour options are new. There used to be colour pickers in its place, but I had already thought about giving default palette options based on feedback, which I think is a much better option. It's convenient for users, and it's good for me, from a creative control perspective.
